oml/datasets/images.py

```diff
@@ -157,8 +157,17 @@ def __init__(
 
     @staticmethod
     def _read_bytes(path: Union[Path, str]) -> bytes:
-        with open(str(path), "rb") as fin:
-            return fin.read()
+        if isinstance(path, str) and path.startswith(('http://', 'https://')):
+            # If the path is a URL, use requests to fetch the content
+            response = requests.get(path)
```

Comment:
From a performance point of view, it would be more efficient to create a session instance `self.session = self.session or requests.Session()` and make requests using `self.session.get(...)`. Otherwise, for each call of `__getitem__`, you will need to establish a new connection with the remote server (DNS resolution, certificate exchange, etc). Establishing a connection can take a huge amount of time.

I'm not sure about an exact implementation but doing `requests.get` is not a good idea for dataset scenario, because we can reuse a session.